### PR TITLE
Telegraf config file and rpm for varnish.

### PR DESCRIPTION
**Pull request by cyrhano**
Added a telegraf config file for varnish and script to generate the corresponding rpm.

### CODE REVIEW

**Review by kthommandra — COMMENTED** [review]
can you include some notes from testing here

File: Arista/ConfigFiles/telegraf-varnish.conf [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+# Configuration for influxdb server to send metrics to
+[[outputs.influxdb]]
+  ## The full HTTP or UDP endpoint URL for your InfluxDB instance.
+  ## Multiple urls can be specified as part of the same cluster,
+  ## this means that only ONE of the urls will be written to each interval.
+  # urls = ["udp://localhost:8089"] # UDP endpoint example
+  urls = ["http://planck:8086"] # required
+  ## The target database for metrics (telegraf will create it if not exists).
+  database = "ServerStats" # required
```
Comment (kthommandra) [inline Arista/ConfigFiles/telegraf-varnish.conf:13]: did you want this to be devStats ?

File: Arista/ConfigFiles/telegraf-varnish.conf [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+
+###############################################################################
+#                            INPUT PLUGINS                                    #
+###############################################################################
+
+# Read metrics about varnish
+[[inputs.varnish]]
+  ## If running as a restricted user you can prepend sudo for additional access:
+  use_sudo = true
```
Comment (kthommandra) [inline Arista/ConfigFiles/telegraf-varnish.conf:32]: is this needed?

File: Arista/ConfigFiles/telegraf-varnish.conf [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+  ## By default, telegraf gather stats for 3 metric points.
+  ## Setting stats will override the defaults shown below.
+  ## Glob matching can be used, ie, stats = ["MAIN.*"]
+  ## stats may also be set to ["*"], which will collect all stats
+  stats = ["MAIN.cache_hit", "MAIN.cache_miss", "MAIN.uptime"]
+
+  ## Optional name for the varnish instance (or working directory) to query
+  ## Usually appened after -n in varnish cli
+  # instance_name = instanceName
```
Comment (kthommandra) [inline Arista/ConfigFiles/telegraf-varnish.conf:45]: You mentioned that there may be additional users for varnish stats.
Would instanceName be used to distinguish these?

File: Arista/ConfigFiles/telegraf-varnish.conf (hunk 1 above)
Comment (cyrhano) [reply Arista/ConfigFiles/telegraf-varnish.conf:13]: We've decided to keep it in ServerStats to be consistent with other non-Arista applications which push stats there.

File: Arista/ConfigFiles/telegraf-varnish.conf (hunk 2 above)
Comment (cyrhano) [reply Arista/ConfigFiles/telegraf-varnish.conf:32]: Yes, telegraf invokes "varnishstat" which is the application that can read stats from varnishd. It needs to be able to read a secret file to do this which is root owned.

File: Arista/ConfigFiles/telegraf-varnish.conf (hunk 3 above)
Comment (cyrhano) [reply Arista/ConfigFiles/telegraf-varnish.conf:45]: There's no need, the hostname is used as a tag so we can distinguish them in that way.

File: Arista/ConfigFiles/telegraf-varnish.conf (hunk 1 above)
Comment (kthommandra) [reply Arista/ConfigFiles/telegraf-varnish.conf:13]: ok

File: Arista/ConfigFiles/telegraf-varnish.conf (hunk 2 above)
Comment (kthommandra) [reply Arista/ConfigFiles/telegraf-varnish.conf:32]: ok

File: Arista/ConfigFiles/telegraf-varnish.conf (hunk 3 above)
Comment (kthommandra) [reply Arista/ConfigFiles/telegraf-varnish.conf:45]: ok

**Review by kthommandra — APPROVED** [review]
LGTM

**Pull request merged**